Lib/test/test_cmd_line.py

```diff
@@ -201,6 +201,27 @@ def test_run_module_bug1764407(self):
         self.assertTrue(data.find(b'1 loop') != -1)
         self.assertTrue(data.find(b'__main__.Timer') != -1)
 
+    @support.cpython_only
+    def test_null_byte_in_interactive_mode(self):
+        # gh-140594: heap-buffer-underflow в PyOS_StdioReadline при \0 в интерактивном вводе
+        env = os.environ.copy()
+        env.pop('PYTHONSTARTUP', None)
+        args = [sys.executable, '-I', '-S', '-q', '-i']
+        p = subprocess.Popen(
+            args,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        out, _ = p.communicate(b'\x00', timeout=10)
+        self.assertEqual(
+            p.returncode, 0,
+            msg=f"Interpreter aborted on NUL input, output:\n{out[:500]!r}"
+        )
+        if out:
```

Comment:
What is this? why are we checking for the address sanitizer error? running the test with ASAN should be sufficient to avoid having the bug.